### PR TITLE
zstream trigger: fix trigger logic, add logs and test

### PR DESCRIPTION
**Pull request by myakove**
Fix logic of trigger job when we have new zstream version.
Add more logs.
Add a test to cover new zstream version trigger.

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest build-container`: Retest build-container

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

**Comment by redhat-qe-bot2** [comment]
New container for quay.io/redhat_msi/ci-jobs-trigger:pr-84 published

File: ci_jobs_trigger/tests/zstream_trigger/test_zstream_trigger.py [hunk 1]
```diff
@@ -66,7 +74,9 @@ def test_process_and_trigger_jobs(mocker, config_dict):
         GET_ACCEPTED_CLUSTER_VERSIONS_PATH,
         return_value=VERSIONS,
     )
-    mocker.patch(TRIGGER_JOBS_PATH, return_value=True)
+    openshift_ci_trigger_job_mocker = mocker.patch(TRIGGER_JOBS_PATH)
+    openshift_ci_trigger_job_mocker.ok = True
```
Comment (rnetser) [inline ci_jobs_trigger/tests/zstream_trigger/test_zstream_trigger.py:78]: fixture?

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ci_jobs_trigger/tests/zstream_trigger/test_zstream_trigger.py (hunk 1 above)
Comment (myakove) [reply ci_jobs_trigger/tests/zstream_trigger/test_zstream_trigger.py:78]: No, this is not a fixture.
Usually fixture in pytest will be decorated with `@pytest.fixture`

**Review by rnetser — APPROVED** [review]

**Comment by msi-b1** [comment]
New container for quay.io/redhat_msi/ci-jobs-trigger:pr-84 published

**Review by rnetser — APPROVED** [review]

**Comment by msi-b1** [comment]
New container for quay.io/redhat_msi/ci-jobs-trigger:latest published

**Pull request merged**